### PR TITLE
Additional color primaries

### PR DESCRIPTION
**Pull request by kodabb**
Hi,
I've added some color primaries used in old files or in very recent ones.
Please let me know if you would like me to change anything.
Thanks

### CODE REVIEW

**Comment by sekrit-twc** [comment]
These can all be combined into a single commit. Can you provide a reference for the numbering of JEDEC-P22 and any relevant test data?

**Comment by kodabb** [comment]
Commits squashed and build fixed.

Regarding jedec-p22, it is defined as such in 23001-8_2013, while h264/5.pdf name it EBU Tech. 3213-E (1975). I do not know why they jumped from 12 to 22. I can create a test sample if you wish.

File: src/testapp/table.cpp [hunk 1]
```diff
@@ -85,13 +85,18 @@ const zimg::static_string_map<TransferCharacteristics, 6> g_transfer_table{
...
+	{ "470_bg",    ColorPrimaries::REC_470_BG },
 	{ "smpte_c",   ColorPrimaries::SMPTE_C },
 	{ "709",       ColorPrimaries::REC_709 },
+	{ "film",      ColorPrimaries::FILM },
 	{ "2020",      ColorPrimaries::REC_2020 },
+	{ "st428",     ColorPrimaries::ST_428 },
 	{ "dcip3",     ColorPrimaries::DCI_P3 },
-	{ "dcip3_d65", ColorPrimaries::DCI_P3_D65 }
+	{ "dcip3_d65", ColorPrimaries::DCI_P3_D65 },
+	{ "jedec-p22", ColorPrimaries::JEDEC_P22 },
```
Comment (sekrit-twc) [inline src/testapp/table.cpp]: use underscores

File: src/zimg/api/zimg.h [hunk 2]
```diff
@@ -293,11 +293,16 @@ typedef enum zimg_color_primaries_e {
...
+	ZIMG_PRIMARIES_470_BG      = 5,
 	ZIMG_PRIMARIES_170M        = 6,
 	ZIMG_PRIMARIES_240M        = 7,  /* Equivalent to 6. */
+	ZIMG_PRIMARIES_FILM        = 8,
 	ZIMG_PRIMARIES_2020        = 9,
+	ZIMG_PRIMARIES_ST428       = 10,
 	ZIMG_PRIMARIES_ST431_2     = 11,
-	ZIMG_PRIMARIES_ST432_1     = 12
+	ZIMG_PRIMARIES_ST432_1     = 12,
+	ZIMG_PRIMARIES_JEDEC_P22   = 22,
```
Comment (sekrit-twc) [inline src/zimg/api/zimg.h]: The naming should follow ITU-T H.264/5: ZIMG_PRIMARIES_EBU3213_E. Also, no trailing comma per C89.

File: src/zimg/colorspace/colorspace_param.h [hunk 3]
```diff
@@ -20,14 +20,21 @@ constexpr double REC_2020_KR = 0.2627;
...
 constexpr double REC_2020_PRIMARIES[3][2] = { { 0.708, 0.292 }, { 0.170, 0.797 }, { 0.131, 0.046 } };
 constexpr double DCI_P3_PRIMARIES[3][2] = { { 0.680, 0.320 } , { 0.265, 0.690 }, { 0.150, 0.060 } };
+constexpr double ST_428_PRIMARIES[3][2] = { { 0.735, 0.265 }, { 0.274, 0.718 }, { 0.167, 0.009 } };
+constexpr double JEDEC_P22_PRIMARIES[3][2] = { { 0.630, 0.340 }, { 0.295, 0.605 }, { 0.155, 0.077 } };
 
 // White points in XY.
+constexpr double ILLUMINANT_C[2]   = { 0.31, 0.316 };
 constexpr double ILLUMINANT_DCI[2] = { 0.314, 0.351 };
 constexpr double ILLUMINANT_D65[2] = { 0.3127, 0.3290 };
+constexpr double ILLUMINANT_E[2]   = { 1 / 3.0f, 1 / 3.0f };
```
Comment (sekrit-twc) [inline src/zimg/colorspace/colorspace_param.h]: 1.0 / 3.0

**Comment by kodabb** [comment]
updated following review

**Comment by sekrit-twc** [comment]
Approved.

PLEASE NOTE: You must agree to either assign copyright of code to the project owner or license contributions under WTFPL (see COPYING).

**Comment by kodabb** [comment]
I agree to license my contributions under the WTFPL lincense.

**Comment by ValZapod** [comment]
> numbering of JEDEC-P22 and any relevant test data

There is none. Only this big document. http://www.bitsavers.org/components/rca/crt/TPM-1508A_RCA_Phosphors_Oct61.pdf and some books on libgen. I suppose the whole derivation of primaries is a crazy secret or something. I was not able to find BT.709 primaries too, only reference to CCIR Rec. XA/11 1986-90f, a.k.a. IWP 11/6-2108 (Canada).

> 3213-E (1975).

There is nothing there that defines JEDEC P22. It is actually PAL standard. IMHO, just a typo in ITU-T H.273.

**Comment by ValZapod** [comment]
Why the commit? As I said there is nothing that uses those primaries in 3213-E (or its supplement). FFmpeg was also confused about it, here: 

https://patchwork.ffmpeg.org/comment/43278/

**Comment by sekrit-twc** [comment]
Internal cleanup.

**Pull request merged**